### ISSUE TITLE
too many dependencies

### ISSUE THREAD

**Issue by Diluka** [nestjs/terminus#561]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

install all the packages

## Expected behavior

do not install the packages I don't need

## What is the motivation / use case for changing the behavior?

```json
{
    "@nestjs/terminus": {
      "version": "6.5.6",
      "resolved": "https://registry.npm.taobao.org/@nestjs/terminus/download/@nestjs/terminus-6.5.6.tgz",
      "integrity": "sha1-6XejwZG5NrbZGVyrQ/rYEslykXA=",
      "requires": {
        "@grpc/proto-loader": "0.5.3",
        "@nestjs/microservices": "6.11.6",
        "@nestjs/mongoose": "6.3.1",
        "@nestjs/platform-express": "6.11.6",
        "@nestjs/platform-fastify": "6.11.6",
        "@nestjs/typeorm": "6.2.0",
        "amqp-connection-manager": "3.2.0",
        "amqplib": "0.5.5",
        "check-disk-space": "2.1.0",
        "grpc": "1.24.2",
        "mongoose": "5.8.11",
        "mqtt": "3.0.0",
        "nats": "1.4.0",
        "redis": "2.8.0",
        "typeorm": "0.2.22"
      },
}
```

## Environment

<pre><code>
Nest version: 6.5.6

 
For Tooling issues:
- Node version:  v12.14.1 
- Platform:  Linux 

Others:

</code></pre>

**Comment by BrunnerLivio** [comment]
These dependencies are declared as `optionalDependenices`. As recommended in the [Installation](https://github.com/nestjs/terminus#installation), we recommend using the `--no-optional` install option. Is that flexibility good enough?

```bash
➜ npm i @nestjs/terminus &&  tree -L 1 node_modules
node_modules
├── check-disk-space
├── d
├── es5-ext
├── es6-iterator
├── es6-symbol
├── ext
├── @nestjs
├── next-tick
└── type
```

Though, thanks to your issue, I've found that `check-disk-space`-package is actually a direct dependency, even though it should be optional too. Will fix that asap.

**Comment by Diluka** [comment]
My suggestion is remove all the `optionalDependenices` and make a list of what feature need what libs. And this instrument should also be [here](https://docs.nestjs.com/recipes/terminus).

It needs to keep `--no-optional` in every `npm install` commands. It not good enough.

Like `@nestjs/platform-express` and `@nestjs/platform-fastify` they are not `optionalDependenices` to `@nestjs/core`

**Comment by BrunnerLivio** [comment]
@Diluka Sorry for the late reply.

So there are two options to approach this:
- Distribute health check amongst packages (e.g. `@nestjs/terminus-disk` or `@nestjs/terminus-microservices` package)
- Remove all the `optionalDependencies` and lazy load the required dependencies instead. This could cause issues with webpack, but better developer experience.

@kamilmysliwiec how did you approach this regarding webpack? Any insights?

**Comment by Diluka** [comment]
@BrunnerLivio one or more libs i use don't support webpack

**Comment by BrunnerLivio** [comment]
Resolved in [7.0.0](https://github.com/nestjs/terminus/releases/tag/7.0.0) :tada:
